tests/test_project_state.py

```
@@ -0,0 +1,100 @@
+
+def test_file_as_directory(tmp_path):
+    """Test that passing a file instead of directory raises error."""
+    test_file = tmp_path / "test.txt"
+    test_file.write_text("test")
+    
+    with pytest.raises(ProjectStateError):
+        is_new_project(str(test_file))
+
+
+@pytest.mark.skipif(os.name == "nt", reason="Permission tests unreliable on Windows")
+def test_permission_error(tmp_path):
```

Ruff (0.8.2)

Missing return type annotation for public function `test_permission_error`

Add return type annotation: `None`

(ANN201)